XIPs/xip-16-image-url-content-type.md

```diff
@@ -0,0 +1,54 @@
+---
+xip: 16
+title: Image Url Content Type
+description: Content type for image url
+author: JsonPreet
+status: Review
+type: Standards Track
+category: XRC
+created: 2023-02-17
+---
+
+## Abstract
+
+Based on [XIP-5](https://github.com/nakajima/XIPs/blob/patch-1/XIPs/xip-5-message-content-types.md) this XRC proposes a new content type xtmp.org/media-image-url:1.0 for Media Image Url only, that do not consist of text. For this content type developers have to send External Image url as string. No extra parameters required. 
+
+## Motivation
+
+As per users requirements, and for my platform I decided to create a new content type for XMTP which also help to other developers for their apps. We can send any type Image Url and it supports GIF url also.
+
+## Specification
+
+Proposed content type:
+
+```js
+{
+  authorityId: "xmtp.org"
+  typeId: "media-image-url"
+  versionMajor: 0
+  versionMinor: 1
+}
+```
+
+
+The content of the encoded message is arbitrary data. The content MUST not be more than 1MB.
```

Comment:
I would think if the content is only a URL, we could safely restrict it to something much smaller than 1MB. Even 10kb would be very generous for most URLs, and longer than what some browsers support. Edge for example limits to ~2k characters.

We can also specify that the URL must be encoded as UTF-8 text or the content would be rejected, which would remove some potential for misuse.

Reply:
Agreed on URL size limit. 

On Encoding I am using standard TextEncoder
```
class ImageCodec {
    get contentType() {
        return ContentTypeImageKey
    }

    encode(content: string): EncodedContent {
        return {
            type: ContentTypeImageKey,
            parameters: {},
            content: new TextEncoder().encode(content),
        }
    }

    decode(content: EncodedContent): string {
        // console.log(content.content.toString())
        const uint8Array = content.content
        const key = new TextDecoder().decode(uint8Array)
        return key
    }
}
```